src/java/org/apache/cassandra/security/FileBasedSslContextFactory.java

```diff
@@ -294,5 +301,38 @@ protected boolean passwordMatchesIfPresent(String keyPassword)
         {
             return StringUtils.isEmpty(password) || keyPassword.equals(password);
         }
+
+        private static String resolvePassword(String keystoreFilePath, String password, String passwordFilePath)
+        {
+            if (password != null)
+                return password;
+
+            if (StringUtils.isEmpty(passwordFilePath))
+                return password;
+
+            File keystorePasswordFile = new File(passwordFilePath);
+
+            if (!keystorePasswordFile.exists())
+            {
+                final String msg = format("keystore password file %s does not exist", keystorePasswordFile.path());
+                throw new ConfigurationException(msg);
+            }
+
+            try
+            {
+                // we expect a password to be on the first line
+                List<String> lines = FileUtils.readLines(keystorePasswordFile);
+                if (lines.isEmpty())
+                    return null;
```

Comment:
@maulin-vasavada here we should return `""`.

Reply:
done